develop.html.mako

```diff
@@ -91,135 +91,162 @@ seeking or providing support.</em></p>
-    requirements, which will help ensure your PRs pass automatic tests.</li>
-
+    <li><a name="pr-requires-issue"></a>
+        <b>Issues for Pull Requests:
+        <em>It is required to submit an issue in the issue tracker before submitting a pull request.</em>
+        </b>
+        It is a frequent occurence that pull requests submitted either don't describe
+        what problem they intend to solve, or they solve the problem in an inappropriate
+        or sub-optimal way.   Having a bug or feature issue opened ahead of time
+        will allow us to to understand the use case and discuss the best approach
+        before the contributor makes the effort to write the code and tests.
+        The issue number used by Github is also linked in the changelog and is used to 
+        generate a changelog file entry for merge.
+    </li>
+    <li>
+        <a name="pr-requires-tests"></a>
+        <b>Always include tests with the change in code</b> - SQLAlchemy does not
+        commit changes to code without a corresponding test - every line of code that
+        isn't tested is technically a bug in itself.  When a pull request is submitted
+        without tests, it <b>cannot be merged</b> until a test is written.  It is
+        extremely common that we receive a pull request that is one line long,
+        yet we need to write many dozens of lines to test it, since <b>99% of the work
+        for any code change is writing the tests</b>.  <b>Please don't submit
+        one-liner PRs without tests.  Only submit a <a href="participate.html#bugs">complete
+        bug report</a> for code changes where you aren't able to provide tests</b>.
+    </li>
+    <li><b>Please follow up for "requests for changes" on your pull request, as well
+        as subscribe to the Gerrit review to receive comments.</b> - Pull requests where
+        requested changes are not responded towards will be closed.
+    </li>
+    <li>Please make sure you are configured to run the current <a href="#environment">Development Environment</a>
+        requirements, which will help ensure your PRs pass automatic tests.
+    </li>
+    <li>Instead of editing the Changelog file itself, please adopt the following workflow:
+        <ul>
+            <li>If the PR does not already have a corresponding issue in GitHub,
+                please create a new issue in GitHub.
+                (See <a href="#pr-requires-issue">Issues for Pull Requests</a>)
+            </li>
+            <li>Create a per-issue changelog file with the path
+                `/doc/build/changelog/unreleased_{Version}/{IssueId}.rst`, in which
+                `IssueId` is the auto-generated GitHub Issue Id.
+            </li>
+            <li>The file format for a changelog file uses the `.rst` markup described at 
+                <a href="https://github.com/sqlalchemyorg/changelog">sqlalchemyorg/changelog</a>. 
+                See the `README.txt` file in the `/doc/build/unreleased_XX` directory 
```

Comment:
doc/build/changelog/unreleased_XX/README.txt is the path

however README.txt doesn't have an example right now

Reply:
This section now links to `#pr-requires-issue` above

I started working on the example; it's in the other repo though so can't be part of this commit.

Reply:
ok can you push out the fix to that path i can merge and publish

Reply:
@zzzeek a few questions.

I am making an actual `unreleased_XX/README.txt` folder/file for the example in the sqlalchemy repo. I can redo that for something here.

That file has two sections. The first is a basically the existing README.txt from versioned. The second section is a quick example and overview.

I noticed the .rst files tend to use `:tickets:` but the https://github.com/sqlalchemyorg/changelog/blob/master/README.rst notes `:ticket:`

Looking at the code, it seems most of the per-ticket items .rst files use :tickets: but the generated files typically use :ticket:

Are these interchangable?  Am I overthinking this?

Reply:
it's supposed to be `:tickets:` because there can be mutiple present there.

There's a README.txt already present, we can add to that?  https://github.com/sqlalchemy/sqlalchemy/blob/master/doc/build/changelog/unreleased_13/README.txt    the unreleased_XX folder is there too already?

Reply:
I added a new `unreleased_xx` folder, and used the README from the 1.x branches as the starting point. I was thinking about updating the existing README files to reference that one

Reply:
we can call it unreleased_XX here but it means to look in each numbered folder